src/tyck.rs

```diff
@@ -6,38 +9,82 @@ pub enum TypeCheckInfo {
 pub trait StaticBase {
     fn type_check(tyck_info: &TypeCheckInfo) -> bool;
 
-    fn type_check_info() -> TypeCheckInfo;
+    fn tyck_info() -> TypeCheckInfo;
+
+    fn lifetime_info() -> RustArgLifetime;
 }
 
-impl StaticBase for i64 {
-    fn type_check(tyck_info: &TypeCheckInfo) -> bool {
+trait LIE<T> {
```

Comment:
Since we'll check about lifetimes inside the casting functions, do we still need such a mechanism, that determines lifetime requirements at the so called "compile time"?

Reply:
In fact we can leave these code here, and go on developing the rest part of language. We can remove code at any time we want, but at least, keep these as is as a prove for the "possibility" of compile time check.

Reply:
**[JUG]** `LIE` is not comprehensive enough